src/Forms/withForm.js

```diff
@@ -0,0 +1,4 @@
+import { hoistStatics, getContext } from 'recompose';
+import { FORM_CONTEXT } from './Form';
+
+export default hoistStatics(getContext(FORM_CONTEXT));
```

Comment:
I've been wondering about how to create "good" HoCs. I remembered from [this blog post about render props](https://cdb.reacttraining.com/use-a-render-prop-50de598f11ce), that writing them takes a log of "ceremony". The post links to [react-router's HoC](https://github.com/ReactTraining/react-router/blob/f77440ec9025d463c6713039ab1a6db1faca99bb/packages/react-router/modules/withRouter.js) as an example.

You can see that they take care of adding `displayName`, `wrappedComponent`, and `propTypes` statics to their HoC. The reason, in particular for the former two props, I think is to provide a better developer experience. Otherwise components don't have names in the React dev tools etc. I've seen this when working on the dashboard a number of times and I think having proper names etc. would be helpful.

So, long story short, how do you feel about us taking care of these things for all HoCs going forward or switching to the render prop pattern altogether?

Reply:
I mean, this higher order component only has the purpose of exposing context so I'm not sure whether all the ceremony is warranted when you can accomplish this in a super simple way using recompose?

Maybe for more complicated stuff? I'm just not sure what value you'd get when you just want to simply pass the context as a property to the children.

Maybe you can give me an example in code of how you think this would work better.